src/game/engine/internals/Rigidbody.js

```diff
@@ -0,0 +1,61 @@
+import { Vector } from '../../shared';
+
+const GRAVITY_ACCELERATION = 0.001;
+
+class Rigidbody {
+  resultVector = Vector.Zero;
+
+  gravityVector = Vector.Zero;
+
+  velocityVector = Vector.Zero;
+
+  accelerationVector = Vector.Zero;
+
+  gravityInit = 0;
+
+  velocityX = 0;
+
+  velocityY = 0;
```

Comment:
Chyba nie ma sensu zbytnio to wyodrębniać od `velocityVector`, możesz całość liczyć tylko na wektorach i powinno być prościej, tzn. np. tam niżej w obliczeniach zamienić
```JS
this.velocityX += this.accelerationVector.x;
this.velocityY += this.accelerationVector.y;
```
na
```JS
this.velocityVector = this.velocity.add(this.accelerationVector);
```

Tak samo `addVelocity` powinno w sumie stać się wtedy pojedynczą linijką.